core/core.go

```diff
@@ -65,7 +69,7 @@ type Driver interface {
 type NetworkDriver interface {
 	// A network driver implements the programming logic for network
 	Driver
-	Init(config *Config, stateDriver StateDriver) error
+	Init(config *Config, info *InstanceInfo, stateDriver StateDriver) error
```

Comment:
@mapuri 

> That way we can fold the StateDriver argument into InstanceInfo as well. Going forward the InstanceInfo structure can then provide a way to add mroe params (if any) that may need to be passed to drivers. WDYT?

It would work too, because StateDriver is on specific InstanceInfo. On other note, should we call it `Config` and `Runtime` to keep it consistence with rest of the terminology.